extensions_userprovided/raspiBackupDiscordWrapper.sh

```diff
@@ -0,0 +1,241 @@
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#######################################################################################################################
+
+set -euf -o pipefail
+
+MYSELF=${0##*/}
+MYNAME=${MYSELF%.*}
+VERSION="0.0.1"
+
+set +u;GIT_DATE="$Date$"; set -u
+GIT_DATE_ONLY=${GIT_DATE/: /}
+GIT_DATE_ONLY=$(cut -f 2 -d ' ' <<< $GIT_DATE)
+GIT_TIME_ONLY=$(cut -f 3 -d ' ' <<< $GIT_DATE)
+set +u;GIT_COMMIT="$Sha1$";set -u
+GIT_COMMIT_ONLY=$(cut -f 2 -d ' ' <<< $GIT_COMMIT | sed 's/\$//')
+
+GIT_CODEVERSION="$MYSELF $VERSION, $GIT_DATE_ONLY/$GIT_TIME_ONLY - $GIT_COMMIT_ONLY"
+
+
+# add pathes if not already set (usually not set in crontab)
+
+if [[ -e /bin/grep ]]; then
+   PATHES="/bin /sbin /usr/bin /usr/sbin /usr/local/bin /usr/local/sbin"
+   for p in $PATHES; do
+      if ! /bin/grep -E -q "[\^:]$p[:$]" <<< $PATH; then
+         [[ -z $PATH ]] && export PATH=$p || export PATH="$p:$PATH"
+      fi
+   done
+fi
+
+
+function readVars() {
+	if [[ -f /tmp/raspiBackup.vars ]]; then
+		source /tmp/raspiBackup.vars						# retrieve some variables from raspiBackup for further processing
+# now following variables are available for further backup processing
+# BACKUP_TARGETDIR refers to the backupdirectory just created
+# BACKUP_TARGETFILE refers to the dd backup file just created
+	else
+		echo "/tmp/raspiBackup.vars not found"
+		exit 42
+	fi
+}
+
+
+
+#### DISCORD SECTION
+
+# read config file
+source /usr/local/etc/discordWrapper.conf
+
+###########################################
+### EXPECTED CONFIG FILE CONTENT
+#
+# URL of the Discord WebHook where to send the message. Adapt to your case
+# CONFIG_DISCORD_URL="https://discord.com/api/webhooks/1234567890.../abcdef...."
+#
+# Possible Colors to be used for Discord Embed message.
+# COLOR_BLUE=255
+# COLOR_RED=16711680
+# COLOR_GREEN=65280
+#
+#
+# Color to be used depending on the raspiBackup execution result: success or failure
+# CONFIG_COLOR_DEFAULT=$COLOR_BLUE
+# CONFIG_COLOR_SUCCESS=$COLOR_GREEN
+# CONFIG_COLOR_FAILURE=$COLOR_RED
+#
+#
+##########################################
+
+DISC_CONTENT=""
+DISC_TITLE=""
+DISC_DESC=""
+DISC_COLOR=$CONFIG_COLOR_DEFAULT
+
+###########################################
+### Default JSON message for Discord notification with a colored embed
+###########################################
+DEFAULTJSON=$(cat<<EOS
```

Comment:
Nice way to initialize a bash variable with a here doc. I prefer to use read (see [this page](https://stackoverflow.com/questions/1167746/how-to-assign-a-heredoc-value-to-a-variable-in-bash)) :-)

Reply:
This is to preserve formatting